swim/labels.go

```diff
@@ -1,5 +1,48 @@
 package swim
 
+import (
+	"errors"
+	"strings"
+)
+
+var (
+	// labelsBannedKeyCharacters keeps a list of all banned occurances for label keys
+	labelsBannedKeyCharacters = []string{
+		// the equal sign is used in label checksum calculation. To prevent
+		// unintended or forced collisions on the checksums of labels we do not
+		// allow the equal sign to be used in the key portion of a label
+		"=",
```

Comment:
`=` seems like too useful character to blacklist. Maybe we can replace it with `\0` instead?

Or rethink the checksum calculation so that we do not need to blacklist any of the characters? E.g. use `concat(hash(hash(key),hash(value)))`, and xor these instead of concatenating `key` and `value`?